### PR TITLE
Added tutorial and pseudobulk code

### PR DESCRIPTION
**Pull request by nrosed**
In this PR I moved the pseudobulk code from the buddi package to the buddi_analysis repo and added the tutorial.

The main part I would like feedback on is everything in the `tutorial` folder and the notebook `sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb`. I updated `scanpy_sc_preprocess_allSplits-liver` notebook to use the most updated (and hopefully fastest) versions of the pseudobulk generation code. I would like feedback specifically on whether you think this code is easy to build off of since I imagine most people will use this code for their own analyses. Also to make sure there is no "liver" specific methods -- they should all be general methods for reading, writing, and analysis.

Thanks!

### CODE REVIEW

File: sc_preprocessing/__init__.py
Comment (ivichadriana) [inline sc_preprocessing/__init__.py:1]: empty file here

File: .gitignore
Comment (ivichadriana) [inline .gitignore:1]: you want this right?

File: sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb [hunk 1]
```diff
@@ -0,0 +1,1899 @@
+{
```
Comment (ivichadriana) [inline sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>typo/grammar:</p><ul><li>This data &gt;&gt;notebook QCs the Kang et.al. data to make sure everything looks good.</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1343250010'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (ivichadriana) [inline sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>as a general comment: in the other notebook you had a really nice description of what is sample_id, stim, etc. i think it could help follow this NB if you add it here also.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1343250012'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

Comment (ivichadriana) [inline sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>is this QC on a gene?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1343250015'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

Comment (ivichadriana) [inline sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>this is very nice! as a comment, when i have used your code in this format, i typically separate the "long" pseudobulks-making loop from everything else. like, i like to QC and see my cell_order, len_vector beforehand. so i print it in a kernel before the loop so i dont make a bunch of files with the wrong cells. just an FYI.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1343250019'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='32'/>

File: tutorial/1_make_pseudobulks.ipynb [hunk 2]
```diff
@@ -0,0 +1,596 @@
+{
```
Comment (ivichadriana) [inline tutorial/1_make_pseudobulks.ipynb:1]: <p>typo/grammar:</p><p>"In this notebook is Part 1 of a BuDDi..." &gt;&gt;&gt;&gt; "This notebook is Part 1..."</p><p> </p><p>Recommendation: </p><p>-This sentence is key to the notebook methods but hard to follow: In this example we have reference stimulated (female single-cell) data, in this analysis the stimulation is Male:Female maps to CTRL:STIM.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1343250040'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

**Comment by ivichadriana** [comment]
-just a couple typos, a couple files i want to make sure you need, and some minor suggestions on what i think could make the NB's more followable (personal preference).
Looking awesome otherwise!!!

File: sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb (hunk 1 above)
Comment (nrosed) [reply sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>ah, interesting. I will think about making it more modular, thanks for letting me know!</p>

Comment (nrosed) [reply sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>updated in: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/9717583b42fad0318684ccf25adb4513954ca3d3" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/9717583b42fad0318684ccf25adb4513954ca3d3</a></p>

Comment (nrosed) [reply sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>good point, added in: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/9717583b42fad0318684ccf25adb4513954ca3d3" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/9717583b42fad0318684ccf25adb4513954ca3d3</a></p>

Comment (nrosed) [reply sc_preprocessing/2_make_pseudobulks/scanpy_sc_preprocess_allSplits-liver.ipynb:1]: <p>yes, I had a bug before with the wrong genes being used, but this isn't necessary any more, deleted in: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/9717583b42fad0318684ccf25adb4513954ca3d3" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/9717583b42fad0318684ccf25adb4513954ca3d3</a> </p>

File: tutorial/1_make_pseudobulks.ipynb (hunk 2 above)
Comment (nrosed) [reply tutorial/1_make_pseudobulks.ipynb:1]: <p>good point, hopefully clearer now: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/f8169ac1f8988b2ad60db46624ba9ac272a2047b" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/f8169ac1f8988b2ad60db46624ba9ac272a2047b</a></p>

**Comment by nrosed** [comment]
<p>fixed in: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/8adee8f88d91094ebaa7d1d8d30096e223a668b6" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/8adee8f88d91094ebaa7d1d8d30096e223a668b6</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1743904808'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>good point added in : <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/7f611d948e352c16ccffc203f360de726831e931" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/7f611d948e352c16ccffc203f360de726831e931</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1743904818'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>good point added here: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/05a034de4095773412e08dd416a1115785ba0f6a" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/05a034de4095773412e08dd416a1115785ba0f6a</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1743904933'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>fixed here: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/05a034de4095773412e08dd416a1115785ba0f6a" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/05a034de4095773412e08dd416a1115785ba0f6a</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1743904878'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>fixed: <a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/05a034de4095773412e08dd416a1115785ba0f6a" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/05a034de4095773412e08dd416a1115785ba0f6a</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1743904855'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>yes, good point. I added an explanation and a method to make it clearer what I am calculating here</p><p><a href="https://github.com/greenelab/buddi_analysis/pull/2/commits/7a33361984e2d23b4e721ac91947ead2679f26a3" target="_blank">https://github.com/greenelab/buddi_analysis/pull/2/commits/7a33361984e2d23b4e721ac91947ead2679f26a3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/2/discussion/#comment-1743904891'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: .gitignore
Comment (nrosed) [reply .gitignore:1]: yes, its needed for python packages

**Comment by nrosed** [comment]
Thanks @ivichadriana !
